### ISSUE TITLE
PyLance misunderstand return type as an object instead of a class

### ISSUE THREAD

**Issue by hspadim-at-dunamis** [microsoft/pylance-release#1406]
<h3> Environment data </h3>

-   Language Server version: 2021.6.0
-   OS and version: linux x64
-   Python version (and distribution if applicable, e.g. Anaconda):  Python 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0] on linux
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: off

<h3> Expected behaviour </h3>

```python
from typing import NewType, Literal
from pydantic import constr

ExchangeName = NewType(
    "ExchangeName", constr(strip_whitespace=True, min_length=0, max_length=50)
)
```

<h3> Actual behaviour </h3>

```python
from typing import NewType, Literal
from pydantic import constr

# Throw error
# Expected class as second argument to NewType Pylance
ExchangeName = NewType(
    "ExchangeName", constr(strip_whitespace=True, min_length=0, max_length=50)
)
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.6.1, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202161-9-june-2021
